compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -2008,9 +2008,13 @@ private STNode parseParameter(SyntaxKind prevParamKind, boolean isParamNameOptio
                 }
 
                 STToken token = peek();
-                Solution solution =
-                        recover(token, ParserRuleContext.PARAMETER_START, prevParamKind, isParamNameOptional);
-
+                Solution solution;
+                if (prevParamKind == SyntaxKind.DEFAULTABLE_PARAM) {
+                    solution = recover(token, ParserRuleContext.PARAMETER_START_WITHOUT_ASTERISK,
+                                       prevParamKind, isParamNameOptional);
```

Comment:
According to spec, a `defaultable-param` can be followed by an `included-record-param` right. So, do we need to skip asterisk here?

Reply:
This should be allowed. Create a separate issue for this.(https://github.com/ballerina-platform/ballerina-lang/issues/27818)

Reply:
Fixed.